### DISCUSSION TITLE
Autogenerate creates indexes separately on new tables

### DISCUSSION THREAD

**Discussion by ziima** [sqlalchemy/alembic#1057] [Usage Questions] [answered]
I noticed autogenerate to create indexes on new tables separately.

Autogenerate output (batch mode on) for a new table looks like:
```python
def upgrade():
    op.create_table(
        "my_table",
        # ... columns ...
        # ... constraints ...
    )
    with op.batch_alter_table("my_table", schema=None) as batch_op:
        batch_op.create_index(batch_op.f("my_table_status_key"), ["status"], unique=False)
```

While it seems that having the `Index` as part of the `create_table` works the same:
```python
def upgrade():
    op.create_table(
        "my_table",
        # ... columns ...
        # ... constraints ...
        Index(op.f("my_table_status_key"), "status", unique=False)
    )
```

Is there a reason for autogenerate to separate creation of indexes, or is it just an unoptimized part of autogenerate?
Similar difference also applies to dropping tables.

**Comment by zzzeek** [comment, marked as answer]
there's no reason it looks like an implementation artifact that I agree is not ideal, and I would support fixing it for this case.
